### PR TITLE
Add container filters

### PR DESCRIPTION
**Pull request by tedoc2000**

### CODE REVIEW

File: maestro/maestro.py [hunk 1]
```diff
@@ -187,21 +187,27 @@ def _gather_dependencies(self, containers, forward=True):
...
             expand_services (boolean): whether to allow expanding service names
                 to their container instance names. If False and a service name
                 is encountered, the method will throw an exception.
+            container_filter (string): optional pattern to filter container names
         """
+        def do_filter(containers):
+            if not container_filter:
+                return containers
+            return [c for c in containers if fnmatch.fnmatch(container_filter, c)]
+
```
Comment (mpetazzoni) [inline maestro/maestro.py:204]: Why not putting this logic line 218 before the final return? You'd save the nested function and the two call sites lines 207 and 210, in favor of a single place before the final `return sorted(set(result))`.

**Review by mpetazzoni — CHANGES_REQUESTED** [review]
Add change to documentation as well, and unit test too.

File: maestro/maestro.py (hunk 1 above)
Comment (tedoc2000) [reply maestro/maestro.py]: Ack

**Comment by tedoc2000** [comment]
Messed up merge

**Pull request closed without merging**